Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -0,0 +1,131 @@
+#:  * `extract` [`--force`] <formula> <tap> [`--version=`<version>]:
+#:    Looks through repository history to find the <version> of <formula> and
+#:    creates a copy in <tap>/Formula/<formula>@<version>.rb. If the tap is
+#:    not installed yet, attempts to install/clone the tap before continuing.
+#:
+#:    If `--force` is passed, the file at the destination will be overwritten
+#:    if it already exists. Otherwise, existing files will be preserved.
+#:
+#:    If an argument is passed through `--version`, <version> of <formula>
+#:    will be extracted and placed in the destination tap. Otherwise, the most
+#:    recent version that can be found will be used.
+
+require "utils/git"
+require "formulary"
+require "tap"
+
+class BottleSpecification
+  def method_missing(m, *_args, &_block)
+    # no-op
+  end
+end
+
+class Module
+  def method_missing(m, *_args, &_block)
+    # no-op
+  end
+end
+
+class DependencyCollector
+  def parse_symbol_spec(spec, tags)
+    case spec
+    when :x11        then X11Requirement.new(spec.to_s, tags)
+    when :xcode      then XcodeRequirement.new(tags)
+    when :linux      then LinuxRequirement.new(tags)
+    when :macos      then MacOSRequirement.new(tags)
+    when :arch       then ArchRequirement.new(tags)
+    when :java       then JavaRequirement.new(tags)
+    when :osxfuse    then OsxfuseRequirement.new(tags)
+    when :tuntap     then TuntapRequirement.new(tags)
+    when :ld64       then ld64_dep_if_needed(tags)
+    else
+      # no-op
+    end
+  end
+
+  module Compat
+    def parse_string_spec(spec, tags)
+      opoo "'depends_on ... => :run' is disabled. There is no replacement." if tags.include?(:run) && ARGV.debug?
+      super
+    end
+  end
+
+  prepend Compat
+end
+
+module Homebrew
+  module_function
+
+  def extract
+    Homebrew::CLI::Parser.parse do
+      flag   "--version="
+      switch :debug
+      switch :force
+    end
+
+    # Expect exactly two named arguments: formula and tap
+    raise UsageError if ARGV.named.length != 2
+
+    destination_tap = Tap.fetch(ARGV.named[1])
+    odie "Cannot extract formula to homebrew/core!" if destination_tap.core_tap?
+    destination_tap.install unless destination_tap.installed?
+
+    core = CoreTap.instance
+    name = ARGV.named.first.downcase
+    repo = core.path
+    file = core.path/"Formula/#{name}.rb"
```

Comment:
`file = repo/"Formula/#{name}.rb"` although it's worth noting not all repos have this structure (the `Formula` directory is optional)

Reply:
I'm aware that the `Formula` dir isn't required, although I figured that it would be fine since we're only reading from the core tap, which _does_ have the directory. Should I consider generalizing this logic a bit so that reading from other taps (which might not have the `Formula` dir) is possible down the road?

Reply:
@alebcay Yeh, it's fine, might want to note it as a comment though just so others looking at the code are aware 👍